ui/src/criteria/attribute.tsx

```diff
@@ -81,23 +141,149 @@ function AttributeEdit(props: AttributeEditProps) {
   const hintDisplayName = (hint: tanagra.EnumHintValue) =>
     hint.displayName || "Unknown Value";
 
-  const enumHintValues = underlay.entities
+  const attributeFilterHint = underlay.entities
     .find((g) => g.name === underlay.primaryEntity)
-    ?.attributes?.find((attribute) => attribute.name === props.data.attribute)
-    ?.attributeFilterHint?.enumHint?.enumHintValues;
+    ?.attributes?.find(
+      (attribute) => attribute.name === props.data.attribute
+    )?.attributeFilterHint;
+  const enumHintValues = attributeFilterHint?.enumHint?.enumHintValues;
+  const integerBoundsHint = attributeFilterHint?.integerBoundsHint;
 
-  const selectionIndex = (hint: tanagra.EnumHintValue) =>
-    props.data.selected.findIndex(
-      (row) => row.id === hint.attributeValue?.int64Val
-    );
+  console.log(underlay);
+  console.log(attributeFilterHint);
 
-  if (enumHintValues?.length === 0) {
+  if (!isValid(integerBoundsHint) && enumHintValues?.length === 0) {
     return (
       <Typography>
         No information for attribute {props.data.attribute}.
       </Typography>
     );
   }
+
+  if (
+    integerBoundsHint?.max !== null &&
+    integerBoundsHint?.max !== undefined &&
+    integerBoundsHint?.min !== null &&
+    integerBoundsHint?.min !== undefined
+  ) {
+    const minBound = integerBoundsHint?.min;
+    const maxBound = integerBoundsHint?.max;
+
+    const [minInputValue, setMinInputValue] = React.useState<number>(minBound);
+    const [maxInputValue, setMaxInputValue] = React.useState<number>(maxBound);
+    const [minValue, setMinValue] = React.useState<number>(minBound);
```

Comment:
I don't understand why two sets of values are needed here (minInput, min, maxInput, max). I would check whether it's actually necessary, or if just a single minValue, maxValue would suffice.

If it is needed, please add a comment in the code explaining this.

Reply:
The situation is that, the values of input box and slider should be isolated. Because, the slider values will be checked within range at all time, but during the time that user enter the number into the input box, no explicit check is supposed to be performed on the input. This is the behavior in All of Us website.

For instance, our year of birth is from 1930 to 2001. A user would like to type "1940", and mistakenly type "19400". If there is checking, the "19400" will be corrected to "2001". The user will need to reenter "1940" instead of simply delete a "0" at the end. So I used two sets of values to implement this behavior as All of Us.